lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -2327,12 +2327,29 @@ CompilerType
 TypeSystemSwiftTypeRef::GetTypeForFormatters(opaque_compiler_type_t type) {
   return m_swift_ast_context->GetTypeForFormatters(ReconstructType(type));
 }
+
 LazyBool
 TypeSystemSwiftTypeRef::ShouldPrintAsOneLiner(opaque_compiler_type_t type,
                                               ValueObject *valobj) {
-  return m_swift_ast_context->ShouldPrintAsOneLiner(ReconstructType(type),
-                                                    valobj);
+  auto impl = [&]() {
+    if (type) {
+      if (IsImportedType(type, nullptr))
+        return eLazyBoolNo;
+    }
+    if (valobj) {
+      if (valobj->IsBaseClass())
+        return eLazyBoolNo;
+      if ((valobj->GetLanguageFlags() & LanguageFlags::eIsIndirectEnumCase) ==
+          LanguageFlags::eIsIndirectEnumCase)
+        return eLazyBoolNo;
+    }
+
+    return eLazyBoolCalculate;
+  };
+  VALIDATE_AND_RETURN(impl, ShouldPrintAsOneLiner, type,
+                      (ReconstructType(type), valobj));
```

Comment:
This looks sufficiently generic that you could hoist it all the way into the TypeSystemSwift base class and delete the SwiftASTContext implementation.

Reply:
There's nothing TypeSystemSwiftTypeRef-specific.

Reply:
That said, I'm also fine with landing it as is and doing that as an NFC refactor later.